Handling of bytes values in value and key of producer.

MarijnHoum opened the pull request:
Hello,

A PR to fix logging errors when producing bytes to Kafka as values or values. Contains a small unit-test.

I hope you will consider it for inclusion.

Kind regards.

MarijnHoum committed:
Handling of bytes values in value and key of producer.

MarijnHoum committed:
Merge branch 'heysaeid:master' into master

heysaeid reviewed (commented):
I don't understand why we have to do the following 🤔
```
if isinstance(value, bytes):
    value = str('value')
if isinstance(key, bytes):
    key = str('value')
```

MarijnHoum commented:
Hahahah.. and this is why we do reviews..  I'll fix it  ASAP.

MarijnHoum committed:
Properly represent arbitrary bytes when logging 'value' or 'key' of a message..

MarijnHoum commented:
I fixed it and added some tests to check that the transformation works as expected. 

Underlying reason is, of course that key and value can be str, bytes or None in Kafka. However, the json specification doesn't support serializing of bytes, so the logging fails, leaving an ugly logging error.

heysaeid reviewed (changes requested):
Thank you for your attention and cooperation 🙏.

But as a devil's advocate 😈, I suggest you do the byte conversion using a separate method to separate the details from the main work 🤓.

MarijnHoum committed:
Moved formatting logic into formatters.py

heysaeid commented on flask_and_kafka/formatters.py:
```diff
@@ -22,6 +22,9 @@ class ProducerFormatter(logging.Formatter):
 
     def format(self, record):
         super().format(record)
+        for entry_name, entry_value in record.producer_log.items():
```
Logically there is no reason for us to convert all the inputs because the other side might need a byte!! So the best thing is to do the following:
```
def convert_bytes_to_repr(value) -> str :
    if isinstance(value, bytes):
        return repr(value)
    return value

'key': convert_bytes_to_repr(key),
'value': convert_bytes_to_repr(value),
```

MarijnHoum committed:
Do byte->str transformation  only to specific fields.

heysaeid commented:
Thank you @MarijnHoum  🙇 🚀
